shell/platform/darwin/macos/framework/Source/FLEViewController.mm

```diff
@@ -356,6 +361,46 @@ - (BOOL)launchEngineInternalWithAssetsPath:(NSURL*)assets
   flutterArguments.command_line_argv = argv;
   flutterArguments.platform_message_callback = (FlutterPlatformMessageCallback)OnPlatformMessage;
 
+  // Provide data for release mode engine.
+#if NDEBUG
+  const char* vmSnapshotDataPath = [[[NSBundle mainBundle] pathForResource:kVmSnapshotDataPath
```

Comment:
These should all be constructed with `assets` (`[assets  URLByAppendingPathComponent:kFoo]`), not the bundle, since the bundling is done by the application, which passes the path in.